DANS-KNAW/easy-convert-bag-to-deposit
# DD-389: pre-staged.csv replaces payload files

Fixes DD- 389: pre-staged.csv replaces payload files

When applied it will
--------------------
* add pre-staged.csv to the metadata folder of a bag with info from `dd-migration-info-service`
* remove the payload files with the same checksum (as far as unique), don't delete from files.xml
* not implemented for bags with `Is-Version-Of` in `bag-info.txt`

Where should the reviewer @DANS-KNAW/easy start?
------------------------------------------------

How should this be manually tested?
-----------------------------------

* on deasy-2021-03-08.box
  * deposit and publish(!) a new dataset with multiple files
  * create a bag to import on dataverse
    (with up to date compiled/deployed modules `fedora-to-bag`, `bag-to-deposit`, `validate-dans-bag`):

        sudo mv /etc/opt/dans.knaw.nl/easy-convert-bag-to-deposit/ThemathischeCollecties.csv ThemathischeCollecties-bu.csv
        sudo touch /etc/opt/dans.knaw.nl/easy-convert-bag-to-deposit/ThemathischeCollecties.csv
        sudo chmod 777 /etc/opt/dans.knaw.nl/easy-convert-bag-to-deposit/
        rm -rf easy-*.log /vagrant/shared/btf-out/* /vagrant/shared/easy-fedora-to-bag-staging/*
        easy-fedora-to-bag -d 'easy-dataset:17' -o /vagrant/shared/btf-out -f SIP simple # ID of a new manual deposit
        easy-convert-bag-to-deposit -d /vagrant/shared/btf-out/ -t DOI -s FEDORA
        easy-validate-dans-bag /vagrant/shared/btf-out/UUID/UUID # fill in UUIDs with autocompletion

  * ~replace the deasy DOI everywhere in the bag with a published DOI such as https://doi.org/10.17026/dans-x8j-87x3~
  * ~`bagit updatetagmanifests /vagrant/shared/UUID/UUID`~
* on host
  * copy the bag created above: `easy-dtap/shared/btf-out/UUID` to `dd-dtap/shared/UUID`
  * ~~move [war](https://github.com/DANS-KNAW/dataverse/releases/download/v5.5-DANS-DataStation-Sprint-15-r5/dataverse-5.5.war) to `dd-dtap/shared` and make it accessible to user `payara`~~
* once on a fresh dev_archaeology-2021-10-02.box
  * ~~copy/paste API token~~ from https://dar.dans.knaw.nl/dataverseuser.xhtml?selectTab=apiTokenTab
    * `/etc/opt/dans.knaw.nl/dd-dans-deposit-to-dataverse/application.properties`
    * `/etc/opt/dans.knaw.nl/dd-migration-info/application.properties`
  * ~~build/deploy `dd-migration-info` (v1.0.9)~~
  * ~~`./deploy-dataverse-war.sh --from-shared shared-code/dataverse/target/dataverse-5.6.war`~~
  * ~~`./deploy.sh dev_archaeology vagrant configure-cvoc-dataverse`~~
* on dar ([importing-pre-staged-files](https://dans-knaw.github.io/dd-dans-deposit-to-dataverse/#importing-pre-staged-files))
 
      dd-dans-deposit-to-dataverse import -s /vagrant/shared/UUID /vagrant/shared/import-out/
      curl -v -X POST 'http://localhost:20230/datasets/actions/load-from-dataverse'
      curl -v -X GET 'http://localhost:20230/datasets/:persistentId/seq/1/basic-file-metas?persistentId=doi:10.17026/dans-x8j-87x3'

  the 2nd curl mimics the request to `dd-migration-info` by `easy-convert-bag-to-deposit`. The output is used in a unit test.

* Create a fresh bag from the same dataset. The bag should now have `metadata/pre-staged.csv` with content and be valid. 
  
  The first round was done successfully with another version of easy-convert-bag-to-deposit. Requirements for the repeated step:
  * on dar:  `/opt/dans.knaw.nl/dd-migration-info/bin/export-migration-info.sh /vagrant/shared/miginfo.zip`
  * on host (mac),
    * move `dd-dtap/shared/miginfo.zip` to `easy-dtap/shared`
    * copy `/dd-dtap/.dans.knaw.nl-yum-repo/dans.knaw.nl-dd-migration-info`... to `easy-dtap`
    * deploy temporary role: https://github.com/DANS-KNAW/easy-dtap/compare/master...jo-pol:DD-389-pre-staged-miginfo
  * on deasy: `/opt/dans.knaw.nl/dd-migration-info/bin/import-migration-info.sh /vagrant/shared/miginfo.zip`
  * 2nd curl command above returns a json record but `pre-staged.csv` remains empty.
  * In the scenario above _the DOI should be replaced before_ `easy-convert-bag-to-deposit`

_Beyond the scope of this issue_:
The current default content of [account-substitutes.csv](https://github.com/DANS-KNAW/easy-convert-bag-to-deposit/blob/23f0eb93dbc2cfa9cddd78904a0c5b9a1f63eede/src/main/assembly/dist/cfg/account-substitutes.csv#L2) prevents dar to import the deposit. After addressing that issue, importing the the bag of the second round requires import [step 3-5](https://dans-knaw.github.io/dd-dans-deposit-to-dataverse/#importing-pre-staged-files) 


Related pull requests on github
-------------------------------

requires https://github.com/DANS-KNAW/easy-dtap/pull/503
validation https://github.com/DANS-KNAW/easy-validate-dans-bag/pull/107
